src/reblox/grid.rs

```diff
@@ -0,0 +1,595 @@
+
+use std::time::Duration;
+
+use pseudo_random::PseudoRandom;
+
+#[derive(PartialEq,Eq,Clone,Copy)]
+pub enum BlockType {
+    I,
+    J,
+    L,
+    O,
+    S,
+    T,
+    Z,
+    GARBAGE,
+    NONE,
+}
+
+const GRID_WIDTH: u8 = 10;
+const GRID_HEIGHT: u8 = 20;
+const START_BLOCK_SPEED_NANOSEC: u32 = 2000000000;
+const END_BLOCK_SPEED_NANOSEC: u32 = 50000000;
+const BLOCK_SPEED_STEP: u32 =
+    (START_BLOCK_SPEED_NANOSEC - END_BLOCK_SPEED_NANOSEC) / 255;
+const LINES_TO_CLEAR_TO_LVL_UP: u8 = 10;
+
+pub fn p2xy(pos: u8) -> (u8, u8) {
```

Comment:
It is generally considered an anti-pattern to use lonely functions for convertions. You should instead define them on the type itself.

For example, create a type for `pos` and impl `Into<(u8, u8)>`.